front/src/modules/pathfinding/components/Pathfinding/PathfindingV2.tsx

```diff
@@ -0,0 +1,492 @@
+  // const allowances = useSelector(getAllowances, isEqual);
+  const initializerArgs = {
+    origin,
+    destination,
+    rollingStock,
+    pathSteps,
+    pathProperties,
+  };
+  const [pathfindingState, pathfindingDispatch] = useReducer(reducer, initializerArgs, init);
+  const [isInfraLoaded, setIsInfraLoaded] = useState(false);
+  const [reloadCount, setReloadCount] = useState(1);
+  const [isInfraError, setIsInfraError] = useState(false);
+
+  const [isPathfindingInitialized, setIsPathfindingInitialized] = useState(false);
+
+  const [postPathfindingBlocks] =
+    osrdEditoastApi.endpoints.postV2InfraByInfraIdPathfindingBlocks.useMutation();
+  const [postPathProperties] =
+    enhancedEditoastApi.endpoints.postV2InfraByInfraIdPathProperties.useMutation();
+
+  const { data: infra } = osrdEditoastApi.endpoints.getInfraById.useQuery(
+    { id: infraId as number },
+    {
+      refetchOnMountOrArgChange: true,
+      pollingInterval: !isInfraLoaded ? 1000 : undefined,
+    }
+  );
+  const [reloadInfra] = osrdEditoastApi.endpoints.postInfraByIdLoad.useMutation();
+
+  const {
+    updatePathSteps,
+    // updatePowerRestrictionRanges,
+    // updateAllowances,
+  } = useOsrdConfActions();
+
+  useEffect(() => {
+    if (reloadCount <= 5 && infra && infra.state === 'TRANSIENT_ERROR') {
+      setTimeout(() => {
+        reloadInfra({ id: infraId as number }).unwrap();
+        setReloadCount((count) => count + 1);
+      }, 1000);
+    }
+  }, [infra, reloadCount]);
+
+  useEffect(() => {
+    if (infra) {
+      switch (infra.state) {
+        case 'NOT_LOADED': {
+          reloadInfra({ id: infraId as number }).unwrap();
+          setIsInfraLoaded(false);
+          break;
+        }
+        case 'ERROR':
+        case 'TRANSIENT_ERROR': {
+          setIsInfraLoaded(true);
+          break;
+        }
+        case 'CACHED': {
+          setIsInfraLoaded(true);
+          if (isInfraError) setIsInfraError(false);
+          pathfindingDispatch({
+            type: 'INFRA_CHANGED',
+            params: {
+              origin,
+              vias,
+              destination,
+              rollingStock,
+              pathSteps,
+            },
+          });
+          break;
+        }
+        default:
+          break;
+      }
+    }
+  }, [infra]);
+
+  const displayInfraSoftError = () => (
+    <div className="content pathfinding-error my-2">
+      <span className="lead">
+        <Stop />
+      </span>
+      {reloadCount <= 5 ? (
+        <span className="flex-grow-1">{t('errorMessages.unableToLoadInfra', { reloadCount })}</span>
+      ) : (
+        <span className="flex-grow-1">{t('errorMessages.softErrorInfra')}</span>
+      )}
+    </div>
+  );
+
+  const displayInfraHardError = () => (
+    <div className="content pathfinding-error my-2">
+      <span className="lead">
+        <Stop />
+      </span>
+      <span className="flex-grow-1">{t('errorMessages.hardErrorInfra')}</span>
+    </div>
+  );
```

Comment:
Can you move this outside the component Pathfinding, either above or in a new file ?
```suggestion
const InfraSoftError = ({ reloadCount }: { reloadCount: number }) => {
  const { t } = useTranslation(['operationalStudies/manageTrainSchedule']);
  return (
    <div className="content pathfinding-error my-2">
      <span className="lead">
        <Stop />
      </span>
      {reloadCount <= 5 ? (
        <span className="flex-grow-1">{t('errorMessages.unableToLoadInfra', { reloadCount })}</span>
      ) : (
        <span className="flex-grow-1">{t('errorMessages.softErrorInfra')}</span>
      )}
    </div>
  );
};

const InfraHardError = () => {
  const { t } = useTranslation(['operationalStudies/manageTrainSchedule']);
  return (
    <div className="content pathfinding-error my-2">
      <span className="lead">
        <Stop />
      </span>
      <span className="flex-grow-1">{t('errorMessages.hardErrorInfra')}</span>
    </div>
  );
};

```

Reply:
Done. Put them in a new file.